### ISSUE TITLE
Context cancelled (never ending query)

### ISSUE THREAD

**Issue by tkallioras** [arangodb/arangodb#6601]
## My Environment

* __ArangoDB Version__:        3.4 RC1 (enterprise)
* __Storage Engine__:             RocksDB
* __Deployment Mode__:       Cluster
* __Deployment Strategy__:   ArangoDB Starter in Docker
* __Operating System__:        MacOS 10.14
* __Used Package__:              Docker - official Docker library

I have the following kind of documents in a collection named `users`:

```
firstName: "A first name",
lastName: "A last name",
createdTime: "2017-12-13T09:57:06.977764Z"
```

I also have the following query:

```
FOR storedUser IN @@usersCollection
SORT storedUser.createdTime
RETURN storedUser
```

When I run the query through the web interface it never ends (I have only 100 entries). If I run it from the golang driver it returns an error: `context cancelled`

I also tried to create a `persistent index` on `createdTime` field (which is sorted) but I still cannot sort the results based on the `createdTime` field.

If I remove the `SORT` the query runs as expected

**EDIT**: I get the same behavior no matter which field I am trying to sort

**Comment by jsteemann** [comment]
I tried reproducing this locally but for me the query always returns and produces a result. I tried it on Linux though, and also made a few assumptions about the data and setup.

I am wondering if the problem only occurs for specific data or settings.
Can you confirm what `numberOfShards` value you were using for the `users` collection? I tried 1, 2 and 9 shards, and did not have any problem for either setting.
The issue may also be dataset-related. Can you also provide the exact dataset you were using when the problem occurred? 
Regarding indexing: I understood that it happens for you with and without a persistent index. Are there any other indexes present on the collection, and if so, which ones?

I think it would be easiest if you used the "create debug package" button in the web interface to make ArangoDB put together all this information into a zip file, which you could then provide here.
Thanks!

**Comment by tkallioras** [comment]
I have 40 shards and this happens even if I only have the primary index in place. I attach the debug zip here. 
[debugDump.zip](https://github.com/arangodb/arangodb/files/2414832/debugDump.zip)

**Comment by jsteemann** [comment]
Thanks, that dataset/setup triggers the problem here too!
Will now analyze it.

**Comment by jsteemann** [comment]
Will be fixed by https://github.com/arangodb/arangodb/pull/6607, which will become available with the next RC release.

**Comment by jsteemann** [comment]
Fix has been merged back to the 3.4 and devel branches today.

**Comment by sleto-it** [comment]
hi - thanks again for reporting this

we have released 3.4.0-rc which includes the fix for this problem. i am now closing this issue. please comment if needed and we will reopen

thanks again,
